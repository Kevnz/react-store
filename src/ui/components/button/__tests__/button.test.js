import React from 'react'
import 'jest-dom/extend-expect'
import { render, fireEvent } from 'react-testing-library'
import Button from '../button'
describe('The Button component', () => {
  it('should render a button', async () => {
    const { container, getByText } = render(<Button>Click Me</Button>)
    expect(container).toMatchSnapshot()
    expect(container.firstChild).toHaveTextContent('Click Me')
    const btn = getByText('Click Me')
    expect(btn).toMatchSnapshot()
  })

  it('should fire `onClick` when clicked', async () => {
    const mockCallback = jest.fn()
    const { container } = render(
      <Button onClick={mockCallback}>Click Me</Button>
    )
    const btn = container.firstChild
    fireEvent.click(btn)
    expect(mockCallback).toBeCalled()
  })

  it('should render a default button', async () => {
    const { container } = render(<Button>Click Me</Button>)

    expect(container.firstChild).toHaveClass('default')
  })

  it('should render a primary button', async () => {
    const { container } = render(<Button variation="primary">Click Me</Button>)
    expect(container.firstChild).toHaveClass('primary')
  })
})
