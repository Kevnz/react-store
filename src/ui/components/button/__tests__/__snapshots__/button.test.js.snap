// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Button component should render a button 1`] = `
<div>
  <button
    class="default"
  >
    Click Me
    <canvas
      height="0"
      style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
      width="0"
    />
  </button>
</div>
`;

exports[`The Button component should render a button 2`] = `
<button
  class="default"
>
  Click Me
  <canvas
    height="0"
    style="border-radius: inherit; height: 100%; left: 0px; position: absolute; top: 0px; width: 100%;"
    width="0"
  />
</button>
`;
